mara_db/shell.py

```diff
@@ -468,20 +565,39 @@ def __(db: dbs.PostgreSQLDB, target_table: str, csv_format: bool = None, skip_he
     # escape double quotes
     sql = sql.replace('"', '\\"')
 
-    return f'{query_command(db, timezone)} \\\n      --command="{sql}"'
+    return f'{sed_stdin}{query_command(db, timezone)} \\\n      --command="{sql}"'
 
 
 @copy_from_stdin_command.register(dbs.RedshiftDB)
 def __(db: dbs.RedshiftDB, target_table: str, csv_format: bool = None, skip_header: bool = None,
-       delimiter_char: str = None, quote_char: str = None, null_value_string: str = None, timezone: str = None):
+       delimiter_char: str = None, quote_char: str = None, null_value_string: str = None, timezone: str = None,
+       pipe_format: Format = None):
     import uuid
     import datetime
 
     tmp_file_name = f'tmp-{datetime.datetime.now().isoformat()}-{uuid.uuid4().hex}.csv'
     s3_write_command = f'AWS_ACCESS_KEY_ID={db.aws_access_key_id} AWS_SECRET_ACCESS_KEY={db.aws_secret_access_key} aws s3 cp - s3://{db.aws_s3_bucket_name}/{tmp_file_name}'
     s3_delete_tmp_file_command = f'AWS_ACCESS_KEY_ID={db.aws_access_key_id} AWS_SECRET_ACCESS_KEY={db.aws_secret_access_key} aws s3 rm s3://{db.aws_s3_bucket_name}/{tmp_file_name}'
 
-    sql = f"COPY {target_table} FROM 's3://{db.aws_s3_bucket_name}/{tmp_file_name}' access_key_id '{db.aws_access_key_id}' secret_access_key '{db.aws_secret_access_key}'"
+    columns = ''
+    sed_stdin = ''
+    if pipe_format:
+        if isinstance(pipe_format, JsonlFormat):
+            columns = ' (' + ', '.join(['data']) + ')'
+            # escapes JSON escapings since PostgreSQL interprets C-escapes in TEXT mode
+            sed_stdin += "sed 's/\\\\/\\\\x5C/g' \\\n| "
+        elif isinstance(pipe_format, CsvFormat):
+            csv_format = True
+            if delimiter_char is None:
```

Comment:
This pattern is IMO bad: do not allow mixing between old and new way

Reply:
fixed